ui/tests/acceptance/task-fs-test.js

```diff
@@ -37,4 +43,127 @@ module('Acceptance | task fs', function(hooks) {
       'Empty state explains the condition'
     );
   });
+
+  test('visiting /allocations/:allocation_id/:task_name/fs/:path', async function(assert) {
+    const paths = ['some-file.log', 'a/deep/path/to/a/file.log', '/', 'Unicode™®'];
+
+    const testPath = async filePath => {
+      await FS.visitPath({ id: allocation.id, name: task.name, path: filePath });
+      assert.equal(
+        currentURL(),
+        `/allocations/${allocation.id}/${task.name}/fs/${encodeURIComponent(filePath)}`,
+        'No redirect'
+      );
+      assert.equal(FS.breadcrumbsText, `${task.name} ${filePath.replace(/\//g, ' ')}`.trim());
+    };
+
+    await paths.reduce(async (prev, filePath) => {
+      await prev;
+      return testPath(filePath);
+    }, Promise.resolve());
+  });
+
+  test('navigating allocation filesystem', async function(assert) {
+    await FS.visitPath({ id: allocation.id, name: task.name, path: '/' });
+
+    assert.ok(FS.fileViewer.isHidden);
+
+    assert.equal(FS.directoryEntries.length, 4);
+
+    assert.equal(FS.breadcrumbsText, task.name);
+
+    assert.equal(FS.breadcrumbs.length, 1);
+    assert.ok(FS.breadcrumbs[0].isActive);
+    assert.equal(FS.breadcrumbs[0].text, 'task-name');
```

Comment:
A pretty clear downside to the data-in-the-mirage-handler approach is that you can't reach into the mirage db to get the `task-name` value out.

Reply:
agreed. Hard-coding irks me somewhat but feels okay to me for this minimal case. It’s still possible to convert to a more ORMy setup if you think that’s important though!